tests/simulation/test_simulator_itself.py

```diff
@@ -0,0 +1,270 @@
+import secrets
+from typing import List
+
+from hathor.simulator import FakeConnection, Simulator
+from tests import unittest
+
+# these tests will basically setup two independent simulator instances and assert that they generate the same
+# blocks and transactions to the hash throughout the execution
+
+
+class BaseSimulatorSelfTestCase(unittest.TestCase):
+    __test__ = False
+
+    # XXX: these are known not to trigger known issues, this is fine for now because the point of these tests isn't to
+    #      uncover issues from random cases (that is for test_simulator.py), but to test if the simulator itself is
+    #      able to deterministically reproduce the exact same situations
+    sample_seeds: List[int] = [
+    ]
```

Comment:
I wonder if we still benfit from this list. I think we might, otherwise I'd just remove this.

Reply:
I agree we can just remove it.

Reply:
Removed. Thanks!